### ISSUE TITLE
One-Class SVM benign-only filter never works; trained on attack traffic and evaluated with meaningless metrics

### ISSUE THREAD

**Issue by himanshu-45naik** [himanshu-45naik/Network-intrusion-detection#3]
## Problem
`models/oc_svm.py` (line ~22) tries to filter benign-only rows via:

```python
if "Attack Type" in X_train.columns:
    X_train = X_train[X_train["Attack Type"] == 0].drop(columns=["Attack Type"])
```

But by the time this runs, the label column has already been split out in `data_splitting`, and PCA has renamed all features to `PC1..PCn`. The filter never matches, so the OC-SVM silently trains on **all traffic including attacks**, defeating the one-class anomaly-detection premise.

Additionally, the model is evaluated via `mlflow_tracker` against multiclass labels, but `OneClassSVM.predict` returns +1/-1 — so accuracy/precision/recall/F1 computed there are meaningless.

## Fix
- Pass benign-only training data to the OC-SVM explicitly (filter before the label is dropped / before PCA).
- Evaluate as an anomaly detector: map ground truth to ±1 (benign = +1, attack = -1) and compute detection rate / false positive rate.

## Severity
High — model is broken as designed.
